ui/components/content-renderers/footnote-citation.js

```diff
@@ -1,17 +1,77 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 
+import renderNode from '../../util/render-node';
+
 import Link from '../link';
 
+const demoData = {"children":[],"content":[{"content_type":"__text__","text":"The FDCCI was first established by OMB â€œMemo for CIOs: Federal Data\n        Center Consolidation Initiative,â€ issued on February 26, 2010, and\n        modified by subsequent memoranda."}],"depth":3,"identifier":"policy_1__sec_1__para_1__footnote_1","marker":"1","node_type":"footnote","requirement":null,"text":"The FDCCI was first established by OMB â€œMemo for CIOs: Federal Data\n        Center Consolidation Initiative,â€ issued on February 26, 2010, and\n        modified by subsequent memoranda.","type_emblem":"1","policy":{"issuance":"2016-08-01","omb_policy_id":"M-16-19","original_url":"https://www.whitehouse.gov/sites/default/files/omb/memoranda/2016/m_16_19_1.pdf","title":"Data Center Optimization Initiative (DCOI)"}};
 
-export default function FootnoteCitation({ content }) {
-  const href = `#${content.footnote_node}`;
-  return <Link className="footnote-link" href={href}><sup>Footnote { content.text }</sup></Link>;
-}
-FootnoteCitation.propTypes = {
+const propTypes = {
   content: PropTypes.shape({
     footnote_node: PropTypes.string.isRequired,
     text: PropTypes.string.isRequired,
   }).isRequired,
 };
 
+function stateSetter() {
+  return {
+    expanded: false,
+  };
+}
+
+export default class FootnoteCitation extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = stateSetter();
+    this.handleCitationClick = this.handleCitationClick.bind(this);
+  }
+
+  componentDidMount() {
+  }
+
+  componentWillUnmount() {
+  }
+
+  handleCitationClick(e) {
+    e.preventDefault();
+    if (this.state.expanded) {
+      this.setState({ expanded: false });
+    } else {
+      this.setState({ expanded: true });
+    }
+  }
+
+  renderCitation() {
+    return renderNode(demoData);
+  }
+
+  render() {
+    let citation = '';
+    const href = `#${this.props.content.footnote_node}`;
+    const link = (
+      <Link onClick={ (e) => { this.handleCitationClick(e); }}
+        className="footnote-link"
+        href={href}>
+        <sup>Footnote { this.props.content.text }</sup>
+      </Link>
+    );
+    if (this.state.expanded) {
+      citation = (
+        <span>
+          { this.renderCitation() }
+          <span className="linebreaker"></span>
```

Comment:
Could we get the same effect by using the `clearfix` class? Fewer utility classes like this means less we have to remember.